### PR TITLE
tests/gnrc_ipv6_nib: rtr_ltime test

### PR DESCRIPTION
**Pull request by xnumad**
### Contribution description

This PR only provides a test to verify that `_handle_rtr_timeout` does not have any effect on any RA options:
> The Router Lifetime applies only to
                     the router's usefulness as a default router; **it
                     does not apply to information contained in other
                     message fields or options**.  Options that need time
                     limits for their information include their own
                     lifetime fields.
- https://datatracker.ietf.org/doc/html/rfc4861

Emphasis added to highlight what part exactly this test covers. (In particular, it does not cover testing that the "Router Lifetime" has any effect to "the router's usefulness as a default router".)

Test is implemented as part of the "base test function", causing 4 tests to fail if not providing an implementation.

### Issues/PRs references

Test is fulfilled by #20329 and #20371 (alone each, but they are also not mutually exclusive).
While #20329 itself already provides tests, this PR is an _additional_ test (so no duplicate tests across PRs or sth). (In fact, they do not seem to cover the code of that PR; they still succeed if only keeping the tests from the PR.)

### CODE REVIEW

**Comment by xnumad** [comment]
Rebased, meaning PR can now be merged since test does not fail anymore since the code being tested was fixed (through merge of #20329)

**Comment by xnumad** [comment]
Force push: Modified commit: Adjusted for changes meanwhile made to master (via #20329)

File: tests/net/gnrc_ipv6_nib/main.c [hunk 1]
```diff
@@ -1139,40 +1177,42 @@ static void test_handle_pkt__rtr_adv__success(uint8_t rtr_adv_flags,
...
-            nbr_sol = pkt->next->next->data;
-            TEST_ASSERT_EQUAL_INT(ICMPV6_NBR_SOL, nbr_sol->type);
-            TEST_ASSERT_EQUAL_INT(0, nbr_sol->code);
-            TEST_ASSERT(!ipv6_addr_is_multicast(&nbr_sol->tgt));
-            TEST_ASSERT_MESSAGE(ipv6_addr_equal(&_loc_gb, &nbr_sol->tgt),
-                                "_loc_gb != nbr_sol->tgt");
-            /* no further options */
-            TEST_ASSERT_NULL(pkt->next->next->next);
-            gnrc_pktbuf_release(pkt);
+            if (is_first_check) {
```
Comment (fabian18) [inline tests/net/gnrc_ipv6_nib/main.c:1180]: Could you factor this out to a new function, please?
And then not call this inside `test_handle_pkt__rtr_adv__options_success()`.
Effectively it is testing if Duplicate Address Detection is triggered by SLAAC.

**Review by fabian18 — APPROVED** [review]
`make -C tests/net/gnrc_ipv6_nib flash term`

```
OK (53 tests)
{ "threads": [{ "name": "idle", "stack_size": 8192, "stack_used": 436}]}
{ "threads": [{ "name": "main", "stack_size": 12288, "stack_used": 2564}]}
```

**Pull request merged**